### PR TITLE
[MM-55296] Added warning in Workspace dashboard if Mattermost is running as root

### PR DESCRIPTION
**Pull request by gvarma28**
#### Summary

Added "Running as Root" warning under configuration section

Please refer this PR for more context #26130

#### Ticket Link

Jira https://mattermost.atlassian.net/browse/MM-55296
Fixes https://github.com/mattermost/mattermost/issues/25305

#### Release Note

```release-note
NA
```

### CODE REVIEW

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for the PR @gvarma28!

Could you please take a look at the failing test?

```
=== FAIL: channels/api4 TestGetPing/ping_root_status_test/LocalClient (re-run 3) (0.00s)
{"timestamp":"2024-08-16 23:09:39.963 Z","level":"debug","msg":"Able to write to database.","caller":"api4/system.go:188","path":"/api/v4/system/ping","request_id":"uzrf7xkp878oupkyxd4un4kpeh","ip_addr":"","user_id":"","method":"GET"}
{"timestamp":"2024-08-16 23:09:39.963 Z","level":"debug","msg":"Able to write files to local storage.","caller":"filestore/localstore.go:82"}
{"timestamp":"2024-08-16 23:09:39.963 Z","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:182","method":"GET","url":"/api/v4/system/ping","request_id":"uzrf7xkp878oupkyxd4un4kpeh","user_id":"","status_code":"200"}
    system_test.go:94: 
        	Error Trace:	/mattermost/server/channels/api4/system_test.go:94
        	            				/mattermost/server/channels/api4/apitestlib.go:1012
        	Error:      	Not equal: 
        	            	expected: false
        	            	actual  : true
        	Test:       	TestGetPing/ping_root_status_test/LocalClient

=== FAIL: channels/api4 TestGetPing (re-run 3) (0.68s)
```

Thanks!

File: server/channels/api4/system_test.go [hunk 1]
```diff
@@ -83,12 +83,23 @@ func TestGetPing(t *testing.T) {
 		require.Contains(t, respString, "testvalue")
 	}, "ping feature flag test")
 
+	t.Run("ping root_status test", func(t *testing.T) {
```
Comment (hanzei) [inline server/channels/api4/system_test.go:86]: It would be great if the test also confirms that non-system admins don't see this root_status flag.

**Review by hanzei — APPROVED** [review]
LGTM :rocket:

**Review by hanzei — APPROVED** [review]
LGTM :rocket:

File: server/channels/api4/system_test.go [hunk 2]
```diff
@@ -83,12 +83,23 @@ func TestGetPing(t *testing.T) {
...
+		resp, err := th.SystemAdminClient.DoAPIGet(context.Background(), "/system/ping?get_server_status=true", "")
```
Comment (isacikgoz) [inline server/channels/api4/system_test.go:87]: Let's make sure this assertion is made only if the user is `root`.

```suggestion
	t.Run("ping root_status test", func(t *testing.T) {
		if os.Geteuid() == 0 {
			t.Skip("Skipping due to system user not having root privileges")
		}
		resp, err := th.SystemAdminClient.DoAPIGet(context.Background(), "/system/ping?get_server_status=true", "")
```

**Review by isacikgoz — CHANGES_REQUESTED** [review]
Looks good overall, we just need to do the assertion conditionally.

File: server/channels/api4/system_test.go (hunk 1 above)
Comment (gvarma28) [reply server/channels/api4/system_test.go:86]: Added another test with Client `th.Client`. I hope this works.

**Comment by gvarma28** [comment]
Made the suggested changes. 🚀

File: server/channels/api4/system_test.go [hunk 3]
```diff
@@ -83,12 +83,38 @@ func TestGetPing(t *testing.T) {
 		require.Contains(t, respString, "testvalue")
 	}, "ping feature flag test")
 
+	t.Run("ping root_status test", func(t *testing.T) {
+		if os.Geteuid() != 0 {
+			t.Skip("Skipping due to system user not having root privileges")
+		}
```
Comment (hanzei) [inline server/channels/api4/system_test.go:89]: 1/5: I don't think that the test will ever be run as root. It's fine to just check if `root_status` is set for admins.

File: server/channels/api4/system_test.go [hunk 4]
```diff
@@ -83,12 +83,38 @@ func TestGetPing(t *testing.T) {
...
+		resp, err := th.SystemAdminClient.DoAPIGet(context.Background(), "/system/ping?get_server_status=true", "")
```
Comment (hanzei) [inline server/channels/api4/system_test.go:90]: It would be nice, if this test and the others use the `GetPingWithOptions` method instead of `DoAPIGet`.

**Review by isacikgoz — APPROVED** [review]
Fantastic, thanks @gvarma28

**Review by lindy65 — APPROVED** [review]
Thanks @gvarma28 

I'll test this once merged as it's not testable on a spinwick/test server on the PR

cc// @hanzei

**Comment by gvarma28** [comment]
Hi @hanzei, 
I'll be happy to address the requested nitpicks. I can only make the changes in the next 24-36 hours as I'm travelling currently.
If you want to merge this as it is, I'm good with that too. 🙌

**Comment by gvarma28** [comment]
@hanzei I've updated the tests as requested.
Please let me know if I missed anything. Thank you!

File: server/channels/api4/system_test.go (hunk 3 above)
Comment (hanzei) [reply server/channels/api4/system_test.go:85]: We don't need to skip the test any longer, right?

**Review by hanzei — COMMENTED** [review]
Final comment:

https://github.com/mattermost/mattermost/pull/27999#discussion_r1749637464

File: server/channels/api4/system_test.go (hunk 3 above)
Comment (gvarma28) [reply server/channels/api4/system_test.go:85]: Yes, will remove that.

Comment (hanzei) [reply server/channels/api4/system_test.go]: Removed it via 81fa204cb49b18cf6947013bd60520e3670b2976 to move the PR forward

**Comment by hanzei** [comment]
Thank you very much for the contribution @gvarma28 :rocket:

**Pull request merged**